lib/matrix.rb

```diff
@@ -1254,6 +1254,15 @@ def vstack(*matrices)
     self.class.vstack(self, *matrices)
   end
 
+  #
+  # Returns a matrix with every entry being an exponent of e
+  # Matrix[[1, 0], [0, 1]].entrywise_exp
+  #   => Matrix[[2.718281828459045, 1.0], [1.0, 2.718281828459045]]
+
+  def entrywise_exp
+    collect{|e| Math.exp(e)}
```

Comment:
what about calling the variable `component`? `e` is normally the letter used for the exponential and it may be confused.